recipes/genie/all/conanfile.py

```diff
@@ -11,23 +12,31 @@ class GenieConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/bkaradzic/GENie"
     description = "Project generator tool"
-    topics = ("conan", "genie", "project", "generator", "build", "build-systems")
+    topics = ("genie", "project", "generator", "build", "build-systems")
     settings = "os", "arch", "compiler", "build_type"
 
     @property
     def _source_subfolder(self):
         return "source_subfolder"
 
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
     def build_requirements(self):
         if self.settings.compiler == "Visual Studio":
             self.build_requires("cccl/1.1")
 
-        if self.settings.os == "Windows" and tools.os_info.is_windows:
+        if self.settings.os == "Windows" and self._settings_build.os == "Windows":
```

Comment:
Shouldn't this be the following instead?
```suggestion
        if self._settings_build.os == "Windows":
```

Reply:
well, the existing code did `if self.settings.os == "Windows" and tools.os_info.is_windows:` which I interpret as "we are building on windows, for windows", so I think that's what `if self.settings.os == "Windows" and self._settings_build.os == "Windows"` does too,right ?

Reply:
I was thinking about the use case of building on Windows for another os.
But I don't think cross compilers to Linux are used in production.